docs/topics/collections-overview.md

```diff
@@ -287,3 +293,28 @@ The default implementation of `MutableMap` – [`LinkedHashMap`](https://kotlinl
 preserves the order of elements insertion when iterating the map.
 In turn, an alternative implementation – [`HashMap`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-hash-map/index.html) – 
 says nothing about the elements order.
+
+### ArrayDeque
+
+[`ArrayDeque<T`>](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/-array-deque/) is an implementation of a double-ended queue, which allows you to add or remove elements both at the beginning or end of the queue.
+It is a type of collection that uses a resizable `Array`. The `Array` is resized only when it is full.
+
+```kotlin
+fun main() {
+    val deque = ArrayDeque(listOf(1, 2, 3))
+
+    deque.addFirst(0)
+    deque.addLast(4)
+    println(deque) // [0, 1, 2, 3, 4]
+
+    println(deque.first()) // 0
+    println(deque.last()) // 4
+
+    deque.removeFirst()
+    deque.removeLast()
+    println(deque) // [1, 2, 3]
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.4"}
+
+For more information about arrays, see [Arrays](arrays.md).
```

Comment:
This section should not reference the arrays page. Unfortunately, `ArrayDeque` is a "false friend": It's a `Deque` collection, the fact that it is backed by an array is an implementation detail and not relevant to using this collection type. (Similar to an `ArrayList`, it's array-esque nature is, from a usage standpoint, of no concern)